frontend/src/reducers/queries.ts

```diff
@@ -0,0 +1,216 @@
+  setNewRun: boolean,
+  runAutomaticallyTitle:string,
+  setNewName:string,
+  setNewConfirm:boolean,
+  setNewSuccessTriggers:Array<ITrigger>|[],
+  setNewUnSuccessTriggers:Array<ITrigger>|[],
+  queriesAppLength:number,
+  isLoading: boolean,
+  isOpen:boolean,
+  isDuplicate:boolean,
+  waitingQuery:{
+    QueryId?:string,
+    QueryName?:string,
+    QueryCode?:string,
+    QueryTriggers:Array<ITrigger>|[],
+    runAutomatically: boolean,
+    showConfirm: boolean
+  }
+}
+
+const initialState:IQueryState = {
+  queriesApp: [],
+  selectQuery: {
+    selectQueryId: '',
+    selectQueryName: '',
+    selectQueryCode: '',
+    selectQueryTriggers: [],
+    runAutomatically: false,
+    showConfirm: false
+  },
+  setNewCode: '',
+  runAutomaticallyTitle: 'Run query automatically when inputs change',
+  setNewName: '',
+  setNewRun: false,
+  setNewConfirm: false,
+  setNewSuccessTriggers: [],
+  setNewUnSuccessTriggers: [],
+  queriesAppLength: 0,
+  isOpen: false,
+  isDuplicate: false,
+  waitingQuery: {
+    QueryId: '',
+    QueryName: '',
+    QueryCode: '',
+    QueryTriggers: [],
+    runAutomatically: false,
+    showConfirm: false
+  },
+  isLoading: true
+};
+
+export const queries = (state = initialState, action: Routine<any>): IQueryState => {
+  let successTriggers:Array<ITrigger>|[] = [];
+  let UnSuccessTriggers:Array<ITrigger>|[] = [];
+  switch (action.type) {
+    case fetchQueryRoutine.SUCCESS:
+      return {
+        ...state,
+        queriesApp: [...action.payload],
+        queriesAppLength: action.payload.length,
+        isLoading: false
+      };
+    case openQueryRoutine.SUCCESS:
+      const runTitle = action.payload[0].runAutomatically ? 'Run query only when manually triggered'
+        : 'Run query automatically when inputs change';
+      action.payload[0].triggers.find((element:ITrigger) => {
+        if (element.success) {
+          successTriggers = [...successTriggers, element];
+        } return null;
+      });
+      action.payload[0].triggers.find((element:ITrigger) => {
+        if (!element.success) {
+          UnSuccessTriggers = [...UnSuccessTriggers, element];
+        } return null;
+      });
+      return {
+        ...state,
+        selectQuery: {
+          selectQueryId: action.payload[0].id,
+          selectQueryName: action.payload[0].name,
+          selectQueryCode: action.payload[0].code,
+          selectQueryTriggers: action.payload[0].triggers,
+          showConfirm: action.payload[0].showConfirm,
+          runAutomatically: action.payload[0].runAutomatically
+        },
+        setNewCode: action.payload[0].code,
+        setNewName: action.payload[0].name,
+        setNewRun: action.payload[0].runAutomatically,
+        setNewSuccessTriggers: successTriggers,
+        setNewUnSuccessTriggers: UnSuccessTriggers,
+        setNewConfirm: action.payload[0].showConfirm,
+        runAutomaticallyTitle: runTitle
+      };
+    case setSelectQueryRoutine.SUCCESS:
```

Comment:
use destructuring assignment here, they you won't need to write `action.payload...` each time below and the code will become cleaner
```suggestion
    case setSelectQueryRoutine.SUCCESS:
      const { triggers, name, code, ... } = action.payload;
```